src/dex/aave-v3-stata-v2/aave-v3-stata-v2.ts

```diff
@@ -0,0 +1,494 @@
+  }
+
+  // Returns the list of contract adapters (name and index)
+  // for a buy/sell. Return null if there are no adapters.
+  getAdapters(side: SwapSide): { name: string; index: number }[] | null {
+    return null;
+  }
+
+  private _getPoolIdentifier(srcToken: Token, destToken: Token): string {
+    return (
+      this.dexKey +
+      '-' +
+      [srcToken.address.toLowerCase(), destToken.address.toLowerCase()]
+        .sort((a, b) => (a > b ? 1 : -1))
+        .join('_')
+    );
+  }
+
+  // Returns list of pool identifiers that can be used
+  // for a given swap. poolIdentifiers must be unique
+  // across DEXes. It is recommended to use
+  // ${dexKey}_${poolAddress} as a poolIdentifier
+  async getPoolIdentifiers(
+    srcToken: Token,
+    destToken: Token,
+    side: SwapSide,
+    blockNumber: number,
+  ): Promise<string[]> {
+    return [this._getPoolIdentifier(srcToken, destToken)];
+  }
+
+  // Returns pool prices for amounts.
+  // If limitPools is defined only pools in limitPools
+  // should be used. If limitPools is undefined then
+  // any pools can be used.
+  async getPricesVolume(
+    srcToken: Token,
+    destToken: Token,
+    amounts: bigint[],
+    side: SwapSide,
+    blockNumber: number,
+    limitPools?: string[],
+  ): Promise<null | ExchangePrices<AaveV3StataV2Data>> {
+    const src = getTokenType(this.network, srcToken.address);
+    const dest = getTokenType(this.network, destToken.address);
+
+    // one of the tokens must be stata
+    if (![src, dest].includes(TokenType.STATA_TOKEN) || src === dest) {
+      return null;
+    }
+
+    const isSrcStata = src === TokenType.STATA_TOKEN;
+
+    const [stataToken, otherAddressLower] = isSrcStata
+      ? [
+          getTokenFromAddress(this.network, srcToken.address),
+          destToken.address.toLowerCase(),
+        ]
+      : [
+          getTokenFromAddress(this.network, destToken.address),
+          srcToken.address.toLowerCase(),
+        ];
+
+    // the token itself can only swap from/to underlying and aToken, so
+    // - at least one must be stata
+    // - maximum one can be stata
+    // - second one must be underlying or aUnderlying
+    // on the buy side (mint, withdraw) we only support the underlying<->stata conversion, not the aUnderlying
+    if (side === SwapSide.SELL) {
+      if (
+        otherAddressLower !== stataToken.underlying.toLowerCase() &&
+        otherAddressLower !== stataToken.underlyingAToken.toLowerCase()
+      ) {
+        return null;
+      }
+    } else {
+      if (otherAddressLower !== stataToken.underlying.toLowerCase()) {
+        return null;
+      }
+    }
+
+    const stataAddressLower = stataToken.address.toLowerCase();
+
+    if (
+      !this.state[stataAddressLower]?.blockNumber ||
+      blockNumber > this.state[stataAddressLower].blockNumber
+    ) {
+      const cached = await this.dexHelper.cache.get(
+        this.dexKey,
+        this.network,
+        `state_${stataAddressLower}`,
+      );
```

Comment:
It makes RPC call only once per block, considering that this swap is available only for the limited amount of tokens, do we really need to use cache here? just because for me it looks simples to use RPC call and store locally without touching cache at all